### PR TITLE
Add CLI for schema updater

### PR DESCRIPTION
**Pull request by MrCreosote**

### CODE REVIEW

File: src/us/kbase/workspace/database/mongo/MongoWorkspaceDB.java [hunk 1]
```diff
@@ -491,6 +492,8 @@ static void ensureIndexes(final MongoDatabase wsdb) throws CorruptWorkspaceDBExc
 					throw new CorruptWorkspaceDBException(
 							"Found duplicate index keys in the database, " +
 							"aborting startup", dk);
+				} catch (MongoException me) {
+					throw new WorkspaceCommunicationException(ERR_DB_COMM, me);
```
Comment (MrCreosote) [inline src/us/kbase/workspace/database/mongo/MongoWorkspaceDB.java:496]: Discovered this while manually testing the CLI

File: src/us/kbase/workspace/test/kbase/SchemaUpdaterCLITest.java [hunk 2]
```diff
@@ -0,0 +1,271 @@
...
+		runUpdater(new Mocks(), array("file1", "file2"), "", err, 2);
+	}
+	
+	@Test
+	public void noConfigFileFail() throws Exception {
+		final Path file = Paths.get("super_fake_file");
+		final String err = generateHelpString(list(String.format(
+				"Could not read from configuration file super_fake_file: %s " +
+						"(No such file or directory)",
+				file.toAbsolutePath())));
```
Comment (ialarmedalien) [inline src/us/kbase/workspace/test/kbase/SchemaUpdaterCLITest.java:234]: What would the error be if the file exists but cannot be read (e.g. if it has the wrong permissions)?

File: src/us/kbase/workspace/test/kbase/SchemaUpdaterCLITest.java [hunk 3]
```diff
@@ -0,0 +1,271 @@
...
+		runUpdater(m, array(VALID_CONFIG_FILE.toString()), "", err, 2);
+	}
+	
+	@Test
+	public void schemaUpdateFail() throws Exception {
+		final Mocks m = new Mocks();
+		
+		when(m.provider.provide(new KBaseWorkspaceConfig(VALID_CONFIG))).thenReturn(m.client);
+		when(m.updater.update(eq(m.db), any(), eq(false), eq(false)))
+				.thenThrow(new SchemaUpdateException("well kiss my grits"));
```
Comment (ialarmedalien) [inline src/us/kbase/workspace/test/kbase/SchemaUpdaterCLITest.java:267]: 😱

File: src/us/kbase/workspace/kbase/SchemaUpdaterCLI.java [hunk 4]
```diff
@@ -0,0 +1,149 @@
...
+	private SchemaUpdaterCLI(final SchemaUpdater updater, final MongoProvider mongoProvider) {
+		this.updater = updater;
+		this.mongoProvider = mongoProvider;
+	}
+
+	/** Run the CLI.
+	 * @param args the CLI arguments.
+	 */
+	public static void main(final String[] args) {
+		// WARNING - these lines are only tested manually. Retest if you make changes.
```
Comment (ialarmedalien) [inline src/us/kbase/workspace/kbase/SchemaUpdaterCLI.java:90]: ⚠️ ❗ ☠️  😱

Comment (MrCreosote) [reply src/us/kbase/workspace/kbase/SchemaUpdaterCLI.java:90]: Pretty hard to test a function that `System.exit()`s

File: src/us/kbase/workspace/test/kbase/SchemaUpdaterCLITest.java (hunk 2 above)
Comment (MrCreosote) [reply src/us/kbase/workspace/test/kbase/SchemaUpdaterCLITest.java:234]:
```
Could not read from configuration file foo: /home/crushingismybusiness/github/mrcreosote/workspace_deluxe/foo (Permission denied)
Usage: update_workspace_database_schema [-choV] <CONFIG_FILE>
...
```

**Review by ialarmedalien — APPROVED** [review]
Sally forth!

**Pull request merged**